### ISSUE TITLE
[Research] Add DistilHuBERT as Lightweight Voice Emotion Model (23MB, 34ms)

### ISSUE THREAD

**Issue by LakshmiSravyaVedantham** [LakshmiSravyaVedantham/DreamAnalysis#27]
## What

Add **DistilHuBERT** as a lightweight speech emotion recognition model — only **23MB after 8-bit quantization**, processes 4-second utterances in **~34ms** with peak memory under 90MB. This is the Pareto-optimal point for accuracy vs. size in speech emotion.

## Why

- Current emotion2vec+ base: ~90M params, ~360MB, accurate but heavy
- DistilHuBERT: **23MB quantized**, 75.48% WA, 76.40% UA on IEMOCAP
- 34ms per 4-second utterance — matches EEG epoch processing cadence
- Small enough for **mobile/edge deployment** (future React Native app)
- Cross-corpus validation: maintains accuracy across different datasets
- Perfect for the "always-on" voice monitoring path during EEG sessions

### Step 1: Add DistilHuBERT model
```python
# ml/models/voice_emotion_lightweight.py
import torch
from transformers import AutoModelForAudioClassification, AutoFeatureExtractor

class DistilHuBERTEmotionModel:
    """Lightweight voice emotion model for real-time/edge use.
    
    Size: 23MB (int8 quantized)
    Latency: ~34ms per 4-second utterance
    Accuracy: 75.48% WA on IEMOCAP
    Memory: <90MB peak
    """
    def __init__(self, quantize=True):
        self.model = AutoModelForAudioClassification.from_pretrained(
            "superb/hubert-base-superb-er"  # or fine-tuned DistilHuBERT
        )
        if quantize:
            self.model = torch.quantization.quantize_dynamic(
                self.model, {torch.nn.Linear}, dtype=torch.qint8
            )
        self.feature_extractor = AutoFeatureExtractor.from_pretrained(
            "superb/hubert-base-superb-er"
        )
    
    def predict(self, audio_array, sample_rate=16000):
        inputs = self.feature_extractor(audio_array, sampling_rate=sample_rate, return_tensors="pt")
        with torch.no_grad():
            logits = self.model(**inputs).logits
        probs = torch.softmax(logits, dim=-1)
        # Map to 6-class format
        return {"probabilities": mapped_probs, "model": "distilhubert", "latency_ms": elapsed}
```

### Step 2: Model selection strategy
| Context | Model | Latency | Accuracy |
|---------|-------|---------|----------|
| Live WebSocket streaming | DistilHuBERT | 34ms | 75% |
| Real-time with transcription | SenseVoice | 70ms | ~73% |
| Post-session batch analysis | emotion2vec+ | ~500ms | 80%+ |

### Step 3: ONNX export for browser deployment
```python
# Export to ONNX for future browser-side inference
torch.onnx.export(model, dummy_input, "distilhubert_emotion.onnx")
# Enables: voice emotion detection without sending audio to server
```

### Files to Modify
- `ml/models/voice_emotion_lightweight.py` (NEW) — DistilHuBERT wrapper
- `ml/models/voice_emotion_model.py` — Add model selection logic
- `ml/api/routes.py` — Route to appropriate model based on latency requirements

### Expected Impact
- Voice emotion latency: **~500ms → 34ms** for real-time path
- Model size: **360MB → 23MB** (quantized) — 15x smaller
- Enables mobile/edge deployment path
- Memory: <90MB peak — can run alongside EEG processing

## Complexity
LOW — standard HuggingFace transformers, quantization is 3 lines of code

## References
- DistilHuBERT cross-corpus validation (arXiv late 2025)
- HuBERT for emotion: https://huggingface.co/superb/hubert-base-superb-er
- Lightweight SER comparison (arXiv Nov 2025)
